specification/paths/Shipments.json

```diff
@@ -24,7 +24,15 @@
       {
         "name": "filter[search]",
         "in": "query",
-        "description": "String of characters you want to look for in a shipment's attributes. Available fields to look in are:<ul><li>`date_created` (YYYY-MM-DD). <strong>Deprecated</strong> use `filter[date_from]` and `filter[date_to]` instead.</li><li>`description`</li><li>`recipient_address`</li><li>`tracking_code`</li></ul>",
+        "description": "String of characters you want to look for in a shipment's attributes. Available fields to look in are:<ul><li>`date_created` (YYYY-MM-DD). <strong>Deprecated</strong> use `filter[date_from]` and `filter[date_to]` instead.</li><li>`description`</li><li>`customer_reference`</li><li>`recipient_address`</li><li>`tracking_code`</li></ul>",
+        "schema": {
+          "type": "string"
+        }
+      },
+      {
+        "name": "filter[customer_reference]",
+        "in": "query",
+        "description": "Search for a shipment by its customer reference attribute. If a shipments `customer_reference` attribute contains the search string, it is returned in the response.",
```

Comment:
I think we should not search `LIKE` because that's expensive.
My idea was to treat this field as an `id` so it should be an exact match.